### PR TITLE
chore: Test support for JWT auth on third-party API

### PR DESCRIPTION
**Pull request by abbyoung**
# SC-2979

## Proposed changes

Adds test support for api requests to the third-party API to require authorization.

## Reviewer notes

Docs for testing: https://github.com/USSF-ORBIT/ussf-portal/pull/351

### Start the system
```sh
yarn services:up
yarn dev
cd ../ussf-portal-cms
yarn dev
```

Login to the portal http://localhost:3000

### Start storybook

```sh
yarn storybook
```

Login to storybook http://localhost:6006, though the command above should open it for you

---

### As the original developer, I have

- [ ] Met the acceptance criteria
- [ ] Created new stories in Storybook if applicable
- [ ] Created/modified automated unit tests in Jest
- [ ] Created/modified automated [E2E tests](https://github.com/USSF-ORBIT/ussf-portal)
- [ ] Followed guidelines for zero-downtime deploys, if applicable
- [ ] Use [ANDI](https://www.ssa.gov/accessibility/andi/help/install.html) to check for basic a11y issues

### As a reviewer, I have

Check out our [How to review a pull request](https://github.com/USSF-ORBIT/ussf-portal/blob/main/docs/how-to/review-pull-request.md) document.

---

### CODE REVIEW

**Review by gidjin — COMMENTED** [review]
LGTM, I had trouble at the last step, but I think it was the federation servers being unavailable. I'll try again tomorrow.

File: docker-compose.services.yml [hunk 1]
```diff
@@ -51,6 +51,11 @@ services:
     restart: always
     ports:
       - '8080:8080'
+    env_file:
+     # If testing locally with Mobile Connect, set the following
+     # environment variables in your .envrc.local file:
+     # EDIPI, COMMON_NAME, GIVENNAME, SURNAME
+     - .envrc.local
```
[the remark this answers is not in the document]
Comment (abbyoung) [reply docker-compose.services.yml:58]: ooh, i'll give this a try! thanks!

[the remark this answers is not in the document]
Comment (abbyoung) [reply docker-compose.services.yml:58]: i'm getting errors when trying this format in the docker compose file, `Implicit keys need to be on a single line` -- not sure if it's changed with versions.

**Review by gidjin — APPROVED** [review]

File: src/pages/api/thirdparty/auth.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import { NextResponse } from 'next/server'
+import * as jose from 'jose'
+import { NextApiRequest } from 'next'
+// Derived from values in the well-known OpenID configuration
+// https://federation.test.cce.af.mil/oidc/guardian-one/.well-known/openid-configuration
+const JWKS_URI = 'https://federation.test.cce.af.mil/oidc/guardian-one/keys'
+const ISSUER = 'https://federation.test.cce.af.mil/oidc/guardian-one'
```
[the remark this answers is not in the document]
Comment (gidjin) [reply src/pages/api/thirdparty/auth.ts:7]: yep that's what I was thinking too

**Pull request merged**